### ISSUE TITLE
No workshop upload

### ISSUE THREAD

**Issue by ApertureDevelopment** [Meachamp/gmodws#8]
Hello again, I am having a few issues again. Sorry if it takes so long between my messages but I only update my addon every so often.

Here is the log:

```
Loaded.
Starting debug output.
CApplicationManagerPopulateThread took 0 milliseconds to initialize (will have waited on CAppInfoCacheReadFromDiskThread)
CAppInfoCacheReadFromDiskThread took 101 milliseconds to initialize
/data/src/public/tier1/interface.h (264) : Assertion Failed: m_pModule
/data/src/public/tier1/interface.h (264) : Assertion Failed: m_pModule
/data/src/clientdll/steamp2pvoice.cpp (161) : Assertion Failed: s_libVideoFuncs.InitFunctions( LIBVIDEO_LIBRARY_NAME )
/data/src/clientdll/steamp2pvoice.cpp (161) : Assertion Failed: s_libVideoFuncs.InitFunctions( LIBVIDEO_LIBRARY_NAME )
IUGC: 0x1026a10
Set CCheckCallback
Setting Creds: [REMOVED]
sh: 1: /home/teamcity-agent/gmod/gmodws/../ubuntu12_64/gldriverquery: not found
Opted-in Controller Mask for AppId 0: 0
Trying logon state....
STEAMID:76561198090797684
sh: 1: /home/teamcity-agent/gmod/gmodws/../ubuntu12_32/gldriverquery: not found
Logged in.
Starting Item Update.
Update handle: 12447353001590193442
Absolute path: "/home/teamcity-agent/testing/msync2.gma"
Setting Item Content.
Submitting item update.
Call Result: 12447353001590193442

pbFailed: 0
CallResultStatus: 1
m_eResult: 15
m_bUserNeedsToAcceptWorkshopLegalAgreement: 0
IsAPICallCompleted Result: 0
Item update failed (Code 15)
sh: 1: /home/teamcity-agent/gmod/gmodws/../ubuntu12_64/vulkandriverquery: not found


Warning: failed to set thread priority: set failed for 8: -1: setpriority() failed
Warning: failed to set thread priority: set failed for priority 8
Warning: support for elevated priorities is most likely unavailable, suppressing future warnings
Exiting workitem thread
Segmentation fault
```

**Comment by Meachamp** [comment]
This is a strange issue. Code 15 indicates that you're getting access denied, which lines up with the issue we ran into earlier where steam updated the API. However, as of 30 minutes ago, the we've been able to successfully push to the workshop. Are you using the latest version with the bundled steamclient in the releases page?

**Comment by Meachamp** [comment]
This is the current active version we're using. I may have forgotten to post it in releases. If you can confirm this solves your issue, I will do so.
[gmodws.zip](https://github.com/Meachamp/gmodws/files/5360496/gmodws.zip)

**Comment by ApertureDevelopment** [comment]
It deffinetly is a different file then before since it doesnt work at all anymore XD

```
./gmodws: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by ./gmodws)
```

and yes, I have been using the latest release with the buncled steamclient and I also made sure that the SteamCMD can authenticate me

**Comment by Meachamp** [comment]
Can you give me an idea of what glibc version you're running? This binary was built in github's CI environment, which I think is ubuntu 18.04. Anything older than that is going to have issues due to the older version of glibc. I can try giving you a build with an older version linked, but I need to know how old.

**Comment by ApertureDevelopment** [comment]
Well I am using a Up-to-Date version of Debian 10 buster, which makes things a bit complicated, since its the current version of debian but they only very slowly update packages.

and this is my GLibC version: ``(Debian GLIBC 2.28-10) 2.28`` and my GLIBCXX version: ``GLIBCXX_3.4.25``

**Comment by ApertureDevelopment** [comment]
I may also have another idea, but I need to know, how does GModWS know where the steamcmd is located? is there a hard location or something dynamical, maybe a enviroment variable etc ? Because maybe I could build a docker container for it

**Comment by Meachamp** [comment]
It just talks to the steamclient. The steamclient library looks in the home directory of the current user for cached credentials. If you just run gmodws in an ubuntu-18.04 container with the home directory mounted, that'll work. 

Apologies for the delay in response, I've been a bit busy. But I'm going to try building in an old container to get you something with an older glibc.

**Comment by ApertureDevelopment** [comment]
Thank you for that, and I will build ( or at least try to ) build a docker container based on ubuntu to avoid something like that in the future. idk if you would want that dockerfile or not but I would happily PR it if you want

**Comment by Meachamp** [comment]
Perhaps try this? I built it with g++ 8 on debian 10, so hopefully that's old enough to not cause trouble. Let me know.
[gmodws.zip](https://github.com/Meachamp/gmodws/files/5368795/gmodws.zip)

**Comment by ApertureDevelopment** [comment]
Could it be that the changelog parameter has some kind of length limit? Because when I run Gmodws vie my CI it just crashes with "segmentation fault" and when I run it via console without the changelog parameter it finishes successfully

**Comment by ApertureDevelopment** [comment]
Yeah idk what is happening but when running using my CI the binary just instantly exits while when running from ssh it finishes successfully

**Comment by ApertureDevelopment** [comment]
The debug log is also not very helpful, its only this:
```
[21:53:06]	[Step 3/3] Loaded.
[21:53:06]	[Step 3/3] Starting debug output.
[21:53:06]	[Step 3/3] CApplicationManagerPopulateThread took 0 milliseconds to initialize (will have waited on CAppInfoCacheReadFromDiskThread)
[21:53:06]	[Step 3/3] [REMOVED]/custom_script3107495263935673622: line 5: 20412 Segmentation fault      ./gmodws [REMOVED] 2091891205 "[REMOVED]/msync2.gma"
```

And dmsesg is also not saying very much:
```
[1058752.028389] CSteamControlle[21394]: segfault at 0 ip 0000000000000000 sp 00007fe16689fb28 error 14 in gmodws[55c3a23ee000+4000]
[1058752.028461] Code: Bad RIP value.
```

The most strange part is really that it runs when I try it using SSH but fails when I run it trough my CI. Enviroment wise are there no differences between these too

**Comment by ApertureDevelopment** [comment]
btw it seems that github does not use a ubuntu 18.04 container, because the newest avaiable GLIBCXX version for it is .25

**Comment by ApertureDevelopment** [comment]
Okay, so you added the ubuntu-toolchain-r/test ppa to your container, that is the reason why your container has a newer GLIBCXX version then Ubuntu 18.04 should have.

I also have finished building my docker container image for GModWS (ps: it works)

**Comment by Meachamp** [comment]
Apologies for the issues, unfortunately I've had to update the gcc version it builds with since I depend on c++17 features. I think the segfaults you were seeing were because of mismatching library versions. Things get... weird when that happens. However, I did test the binaries on a vanilla 18.04 instance, and it seemed to work fine. Sorry for the trouble - I'll see what I can do to make this a better experience. If you wanted to PR that dockerfile, feel free.
